src/java_bytecode/java_string_library_preprocess.h

```diff
@@ -173,8 +173,15 @@ class java_string_library_preprocesst:public messaget
     symbol_tablet &symbol_table,
     code_blockt &init_code);
 
-  void process_single_operand(
-    exprt::operandst &processed_ops,
+#ifdef CBMC_UNIT_TEST
```

Comment:
I still don't like these `#ifdef`s, but if this has been approved by @peterschrammel I'll surrender (especially because I can't see a better solution at the moment).

Reply:
I think that if we're friending `#ifdef` is unnecessary (it's unlikely we'd create another function `_test` that would be able to access private members of that class).

But having both is not harmful (except readability-wise).